### PR TITLE
Move from RapidXML to Pugixml which supports c++11 features + tvg-id fixes

### PR DESCRIPTION
**Pull request by phunkyfish**
v4.7.1
- Update: Switch from rapixml to pugixml (for C++11 syntax)
- Fixed: tvg-id does not match a channel if no display-names in xmltv
- Fixed: Combine multiple XMLTV channels sharing same id

Pugixml is very similar to rapidxml but supports C++11 syntax allowing cleaner code.

https://github.com/kodi-pvr/pvr.iptvsimple/issues/295

### CODE REVIEW

**Comment by phunkyfish** [comment]
@2small @Sorien @allixx 

Any chance you can test this from a new feature perspective on Kodi Matrix?

Test builds are here for the standard kodi platforms: https://jenkins.kodi.tv/blue/organizations/jenkins/kodi-pvr%2Fpvr.iptvsimple/detail/PR-297/12/artifacts

Just to make sure nothing that was working is broken after changing the underlying XML engine.

**Comment by allixx** [comment]
tested https://github.com/phunkyfish/pvr.iptvsimple/commit/b241214ef025fab919962b3b065643b2db8dea10 with kodi g96737bf on linux x86_64. builds and runs fine (I have a basic channel list without EPG though).
homedir over nfs probably explains a relatively significant load time.

>2019-10-26 12:09:15.681 T:560  NOTICE: AddOnLog: PVR IPTV Simple Client: pvr.iptvsimple - LoadPlayList Playlist Loaded - 211 (ms)
>2019-10-26 12:09:15.681 T:560  NOTICE: AddOnLog: PVR IPTV Simple Client: pvr.iptvsimple - LoadPlayList - Loaded 477 channels.

**Comment by phunkyfish** [comment]
Thanks for testing @allixx !!!

File: src/iptvsimple/data/ChannelEpg.cpp [hunk 1]
```diff
@@ -34,8 +34,11 @@ bool ChannelEpg::UpdateFrom(const xml_node& channelNode, Channels& channels)
     return false;
 
   bool foundChannel = false;
+  bool haveDisplayNames = false;
   for (const auto& displayNameNode : channelNode.children("display-name"))
   {
+    bool haveDisplayNames = true;
```
Comment (ksooo) [inline src/iptvsimple/data/ChannelEpg.cpp]: This doesn't make sense. I guess intention was to assign true to the var declared outside the for loop (line37)?!

**Review by ksooo — CHANGES_REQUESTED** [review]
One bug that needs to be fixed & I hope the new lib was carefully tested to be functionally equivalent to rapidxml.

File: src/iptvsimple/data/ChannelEpg.cpp (hunk 1 above)
Comment (phunkyfish) [reply src/iptvsimple/data/ChannelEpg.cpp]: Oops, so my test would always have passed ;)

Nice find, thanks.

**Comment by phunkyfish** [comment]
I had several users test the latest features/additions with the new lib to make sure everything still worked as expected.

**Review by ksooo — APPROVED** [review]

**Pull request merged**